nixos/modules/programs/wshowkeys.nix

```diff
@@ -0,0 +1,22 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.wshowkeys;
+in {
+  meta.maintainers = with maintainers; [ primeos ];
+
+  options = {
+    programs.wshowkeys = {
+      enable = mkEnableOption ''
+        wshowkeys (displays keypresses on screen on supported Wayland
+        compositors). It requires root permissions to read input events, but
+        these permissions are dropped after startup'';
```

Comment:
Does it work without root if you are part of the `input` group?

Reply:
No, doesn't look like it: https://git.sr.ht/~sircmpwn/wshowkeys/tree/6388a49e0f431d6d5fcbd152b8ae4fa8e87884ee/devmgr.c#L127

But looking at the rest of the code it should be enough to be in the `input` group. It should only require a small patch to detect this and skip the setuid related code.